core/src/mender-client.c

```diff
@@ -670,30 +660,29 @@ mender_client_initialization_work_function(void) {
 
 static mender_err_t
 mender_commit_artifact_data(void) {
-
-    cJSON *json_artifact_name = cJSON_GetObjectItemCaseSensitive(mender_client_deployment_data, "artifact_name");
-    if (NULL == json_artifact_name) {
+    const char *artifact_name;
+    if (MENDER_OK != mender_deployment_data_get_artifact_name(mender_client_deployment_data, &artifact_name)) {
```

Comment:
I'd add `assert(m_c_d_d != NULL)` to the beginning of the function. I know it's not strictly necessary, but it might prevent strange issues in testing.